backend/src/core/decorators/user.decorator.ts

```diff
@@ -0,0 +1,29 @@
+import { createParamDecorator, ExecutionContext } from '@nestjs/common';
+
+/**
+ * User Decorator
+ *
+ * This decorator can be used in both HTTP controllers and WebSocket gateways to fetch user information.
+ * It automatically detects the context (HTTP or WebSocket) and fetches the user accordingly.
+ *
+ * Usage:
+ *
+ * 1. To get the entire user object:
+ *    @User() user: UserEntity
+ *
+ * 2. To get a specific property of the user object (e.g., 'email'):
+ *    @User('email') email: string
+ */
+export const User = createParamDecorator((data: any, ctx: ExecutionContext) => {
+  const type = ctx.getType();
+
+  if (type === 'http') {
+    const request = ctx.switchToHttp().getRequest();
+    return data ? request.user?.[data] : request.user;
+  } else if (type === 'ws') {
+    const client = ctx.switchToWs().getClient();
+    return data ? client.handshake.auth?.[data] : client.handshake.auth?.user;
+  } else {
+    return null;
+  }
+});
```

Comment:
Algumas mudanças pertinentes:
- Adicionar o significado de cada parâmetro nos comentários;
- Criar a variável `user` fora do escopo das condições;
- Correção da forma como o `data` é utilizado no contexto do WS (atualmente, tá tentando ler uma propriedade do `auth`, não do `auth.user`.

O que acha?

```suggestion
import { createParamDecorator, ExecutionContext } from '@nestjs/common';
import { UserEntity } from '../entities';

/**
 * User Decorator
 *
 * This decorator can be used in both HTTP controllers and WebSocket gateways to fetch user information.
 * It automatically detects the context (HTTP or WebSocket) and fetches the user accordingly.
 *
 * @param data Optional. If provided, it will return the specified property of the user object.
 * @param ctx The execution context.
 *
 * Usage:
 *
 * 1. To get the entire user object:
 *    @User() user: UserEntity
 *
 * 2. To get a specific property of the user object (e.g., 'id'):
 *    @User('id') id: number
 */
export const User = createParamDecorator((data: any, ctx: ExecutionContext) => {
  const type = ctx.getType();
  let user: UserEntity;

  if (type === 'http') {
    const request = ctx.switchToHttp().getRequest();
    user = request.user;
  } else if (type === 'ws') {
    const client = ctx.switchToWs().getClient();
    user = client.handshake.auth?.user;
  } else {
    return null;
  }

  return data ? user?.[data] : user;
});

```